code/game/objects/items/food/_food.dm

```diff
@@ -120,3 +120,22 @@
 	AddComponent(/datum/component/germ_sensitive, mapload)
 	if(!preserved_food)
 		AddComponent(/datum/component/decomposition, mapload, decomp_req_handle, decomp_flags = foodtypes, decomp_result = decomp_type, ant_attracting = ant_attracting, custom_time = decomposition_time, stink_particles = decomposition_particles)
+
+/obj/item/food/CheckParts(list/parts, datum/crafting_recipe/food/current_recipe)
+	. = ..()
+	if(!istype(current_recipe))
+		return
+
+	var/made_with_food = FALSE
+	var/final_foodtypes = current_recipe.added_foodtypes
+	for(var/obj/item/food/ingredient in parts)
+		made_with_food = TRUE
+		final_foodtypes |= ingredient.foodtypes
+	if(!made_with_food)
+		return
+	final_foodtypes &= ~current_recipe.removed_foodtypes
+	var/datum/component/edible/edible = GetComponent(/datum/component/edible)
+	//unset the flags so that we can properly replace them with the ones from the ingredients used.
+	edible.foodtypes = NONE
+	REMOVE_TRAIT(src, TRAIT_VALID_DNA_INFUSION, REF(edible))
+	AddComponent(/datum/component/edible, foodtypes = final_foodtypes)
```

Comment:
This getcomp is pretty sad

Can't we hook `COMSIG_ATOM_CHECKPARTS` to handle this in the component itself?

Reply:
All of this will be replaced with a more elegant AddComponentFrom() call once the  other PR is merged.